src/compile.coffee

```diff
@@ -52,35 +57,71 @@ compile = (mediaType, apiElements, filename) ->
+    transitionNo = transitionNo.toValue()
+    httpTransactions = transition.findRecursive('httpTransaction')
+
+    # This gets deleted once we're fully on minim
+    refractTransition = refractTransitions[transitionNo]
+    refractHttpTransactions = children(refractTransition, {element: 'httpTransaction'})
+
     # API Blueprint has a concept of transaction examples and
     # the API Blueprint AST used to expose it. The concept isn't present
     # in API Elements anymore, so we have to detect and backport them.
-    children(parseResult, {element: 'transition'}).map(detectTransactionExamples)
-
-  # Dredd supports only testing of the first request-response pair within
-  # each transaction example. So if we're dealing with API Blueprint, we
-  # iterate over available transactions and skip those, which are not first
-  # within a particular example.
-  #
-  # That's achieved by tracking example number within each transition
-  # and skipping transactions with example number we've already seen.
-  for transition in children(parseResult, {element: 'transition'})
-    example = 0
+    if mediaType is 'text/vnd.apiblueprint'
+      exampleNumbersPerTransaction = detectExampleNumbersPerTransaction(transition)
+    else
+      exampleNumbersPerTransaction = httpTransactions.map( -> 1)
+    hasMoreExamples = Math.max(exampleNumbersPerTransaction...) > 1
+
+    # Dredd supports only testing of the first request-response pair within
+    # each transaction example. So if we're dealing with API Blueprint, we
+    # iterate over available transactions and skip those, which are not first
+    # within a particular example.
+    exampleNo = 0
+    httpTransactions.forEach((httpTransaction, httpTransactionNo) ->
+      httpTransactionNo = httpTransactionNo.toValue()
+      httpTransactionExampleNo = exampleNumbersPerTransaction[httpTransactionNo]
+
+      transactionInfo =
+        refract: refractHttpTransactions[httpTransactionNo]
+        apiElements: httpTransaction
+        exampleNo: if hasMoreExamples then httpTransactionExampleNo else null
 
-    for httpTransaction in children(transition, {element: 'httpTransaction'})
       if mediaType is 'text/vnd.apiblueprint'
-        relevantTransactions.push(httpTransaction) unless httpTransaction.attributes.example is example
+        if httpTransactionExampleNo isnt exampleNo
+          relevantTransactions.push(transactionInfo)
       else
-        relevantTransactions.push(httpTransaction)
-      example = httpTransaction.attributes?.example
+        relevantTransactions.push(transactionInfo)
+
+      exampleNo = httpTransactionExampleNo
+    )
+  )
 
   return relevantTransactions
 
 
+# Detects transaction example numbers for given transition element
+#
+# Returns an array of numbers, where indexes correspond to HTTP transactions
+# within the transition and values represent the example numbers.
+detectExampleNumbersPerTransaction = (transition) ->
```

Comment:
This would be ideally future public interface of `detectTransactionExamples` once it supports minim.